### ISSUE TITLE
error from installation of  torch-scatter

### ISSUE THREAD

**Issue by mgao97** [pyg-team/pytorch_geometric#6756]
### 😵 Describe the installation problem

warnings.warn(ABI_INCOMPATIBILITY_WARNING.format(compiler))
      building 'torch_scatter._segment_coo_cpu' extension
      creating build/temp.linux-x86_64-cpython-38
      creating build/temp.linux-x86_64-cpython-38/csrc
      creating build/temp.linux-x86_64-cpython-38/csrc/cpu
      gcc -pthread -B /conda/envs/env_torch13/compiler_compat -Wno-unused-result -Wsign-compare -DNDEBUG -fwrapv -O2 -Wall -fPIC -O2 -isystem /conda/envs/env_torch13/include -fPIC -O2 -isystem /conda/envs/env_torch13/include -fPIC -DWITH_PYTHON -Icsrc -I/conda/envs/env_torch13/lib/python3.8/site-packages/torch/include -I/conda/envs/env_torch13/lib/python3.8/site-packages/torch/include/torch/csrc/api/include -error: subprocess-exited-with-error
  
  × Running setup.py install for torch-scatter did not run successfully.
  │ exit code: 1
  ╰─> [54 lines of output]
      No CUDA runtime is found, using CUDA_HOME='/usr/local/cuda'
      running install
      /conda/envs/env_torch13/lib/python3.8/site-packages/setuptools/command/install.py:34: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
        warnings.warn(
      running build
      running build_py
      creating build
      creating build/lib.linux-x86_64-cpython-38
      creating build/lib.linux-x86_64-cpython-38/torch_scatter
      copying torch_scatter/utils.py -> build/lib.linux-x86_64-cpython-38/torch_scatter
      copying torch_scatter/__init__.py -> build/lib.linux-x86_64-cpython-38/torch_scatter
      copying torch_scatter/placeholder.py -> build/lib.linux-x86_64-cpython-38/torch_scatter
      copying torch_scatter/segment_csr.py -> build/lib.linux-x86_64-cpython-38/torch_scatter
      copying torch_scatter/segment_coo.py -> build/lib.linux-x86_64-cpython-38/torch_scatter
      copying torch_scatter/scatter.py -> build/lib.linux-x86_64-cpython-38/torch_scatter
      creating build/lib.linux-x86_64-cpython-38/torch_scatter/composite
      copying torch_scatter/composite/__init__.py -> build/lib.linux-x86_64-cpython-38/torch_scatter/composite
      copying torch_scatter/composite/logsumexp.py -> build/lib.linux-x86_64-cpython-38/torch_scatter/composite
      copying torch_scatter/composite/softmax.py -> build/lib.linux-x86_64-cpython-38/torch_scatter/composite
      copying torch_scatter/composite/std.py -> build/lib.linux-x86_64-cpython-38/torch_scatter/composite
      running egg_info
      writing torch_scatter.egg-info/PKG-INFO
      writing dependency_links to torch_scatter.egg-info/dependency_links.txt
      writing requirements to torch_scatter.egg-info/requires.txt
      writing top-level names to torch_scatter.egg-info/top_level.txt
      reading manifest file 'torch_scatter.egg-info/SOURCES.txt'
      reading manifest template 'MANIFEST.in'
      warning: no previously-included files matching '*' found under directory 'test'
      adding license file 'LICENSE'
      writing manifest file 'torch_scatter.egg-info/SOURCES.txt'
      running build_ext
      /conda/envs/env_torch13/lib/python3.8/site-packages/torch/utils/cpp_extension.py:365: UserWarning:
      
                                     !! WARNING !!
      
      !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
      Your compiler (g++ 4.8.5) may be ABI-incompatible with PyTorch!
      Please use a compiler that is ABI-compatible with GCC 5.0 and above.
      See https://gcc.gnu.org/onlinedocs/libstdc++/manual/abi.html.
      
      See https://gist.github.com/goldsborough/d466f43e8ffc948ff92de7486c5216d6
      for instructions on how to install GCC 5 or higher.
      !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
      
                                    !! WARNING !!
I/conda/envs/env_torch13/lib/python3.8/site-packages/torch/include/TH -I/conda/envs/env_torch13/lib/python3.8/site-packages/torch/include/THC -I/conda/envs/env_torch13/include/python3.8 -c csrc/cpu/segment_coo_cpu.cpp -o build/temp.linux-x86_64-cpython-38/csrc/cpu/segment_coo_cpu.o -O3 -Wno-sign-compare -DAT_PARALLEL_OPENMP -fopenmp -DTORCH_API_INCLUDE_EXTENSION_H -DPYBIND11_COMPILER_TYPE=\"_gcc\" -DPYBIND11_STDLIB=\"_libstdcpp\" -DPYBIND11_BUILD_ABI=\"_cxxabi1011\" -DTORCH_EXTENSION_NAME=_segment_coo_cpu -D_GLIBCXX_USE_CXX11_ABI=0 -std=c++14
      gcc: error: unrecognized command line option ‘-std=c++14’
      error: command '/usr/bin/gcc' failed with exit code 1
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: legacy-install-failure

× Encountered error while trying to install package.
╰─> torch-scatter

### Environment

* PyG version:2.2.0
* PyTorch version:1.13.0
* OS:unix
* Python version:3.8
* CUDA/cuDNN version:none
* How you installed PyTorch and PyG (`conda`, `pip`, source):
* Any other relevant information (*e.g.*, version of `torch-scatter`):

**Comment by RaulPPelaez** [comment]
As the error says, you need a more recent version of GCC, at least version 5. Current release is 12.

**Comment by yashpaneliya** [comment]
@RaulPPelaez 

I am also facing the same issue. 

I am using GCC `9.4.0`

torch version: `1.8.0`

Installation command: 
pip install torch-scatter -f https://data.pyg.org/whl/torch-{torch.__version__}.html

**Comment by RaulPPelaez** [comment]
If the issue is the same then you must not be, in fact, using gcc 9.4.0. Depending on how you installed gcc pip might be still using the system one. Perhaps if you show us your env (mamba list) and the actual error output you are getting we could provide further guidance.
In any case, the torch version you are using, 1.8.0 is really old and I would not be surprised if unsupported by gcc 9.4.
I can imagine you are stuck with it for some other dependency issues, though.

**Comment by yashpaneliya** [comment]
Dependencies:

Package                  Version
------------------------ ----------
appdirs                  1.4.4
certifi                  2023.7.22
charset-normalizer       3.2.0
click                    8.1.6
cmake                    3.27.2
docker-pycreds           0.4.0
filelock                 3.12.2
gitdb                    4.0.10
GitPython                3.1.32
idna                     3.4
Jinja2                   3.1.2
joblib                   1.3.2
lit                      16.0.6
MarkupSafe               2.1.3
mpmath                   1.3.0
networkx                 3.1
numpy                    1.25.2
nvidia-cublas-cu11       11.10.3.66
nvidia-cuda-cupti-cu11   11.7.101
nvidia-cuda-nvrtc-cu11   11.7.99
nvidia-cuda-runtime-cu11 11.7.99
nvidia-cudnn-cu11        8.5.0.96
nvidia-cufft-cu11        10.9.0.58
nvidia-curand-cu11       10.2.10.91
nvidia-cusolver-cu11     11.4.0.1
nvidia-cusparse-cu11     11.7.4.91
nvidia-nccl-cu11         2.14.3
nvidia-nvtx-cu11         11.7.91
pandas                   2.0.3
pathtools                0.1.2
pip                      22.0.4
protobuf                 4.24.0
psutil                   5.9.5
pyparsing                3.1.1
python-dateutil          2.8.2
pytz                     2023.3
PyYAML                   6.0.1
requests                 2.31.0
scikit-learn             1.3.0
scipy                    1.11.1
sentry-sdk               1.29.2
setproctitle             1.3.2
setuptools               58.1.0
six                      1.16.0
smmap                    5.0.0
sympy                    1.12
threadpoolctl            3.2.0
torch                    1.8.0
tqdm                     4.66.1
triton                   2.0.0
typing_extensions        4.7.1
tzdata                   2023.3
urllib3                  2.0.4
wandb                    0.15.8
wheel                    0.41.1

Gcc version info:
gcc (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0

Error:
```bash
 https://data.pyg.org/whl/torch-{torch.__version__}.html

Looking in links: https://data.pyg.org/whl/torch-{torch.__version__}.html

Collecting torch-scatter

  Using cached torch_scatter-2.1.1.tar.gz (107 kB)

  Preparing metadata (setup.py) ... done

Building wheels for collected packages: torch-scatter

  Building wheel for torch-scatter (setup.py) ... -
```
Terminal stucks whenever I try to install torch-scatter

**Comment by yashpaneliya** [comment]
Oh, I was not aware of that😅. Let me try again and wait for a while. 
Thanks for the help @RaulPPelaez

**Comment by yashpaneliya** [comment]
> Oh that is not the same error at all then. It is most probably not stuck, when building wheels for this package, such as you are, compiling some files with CUDA is required. This is really slow and might appear as if stuck, you can expect something around 10 mins depending on your system. If you check your running processes you will probably see an instance of either nvcc of cc.
> 
> You can try to call pip with --verbose, or -vv to get more info about what is going on.

Actually, I faced a similar issue mentioned by @mgao97 but then I tried to install a different version of pytorch and then again install torsch-scatter. After that, I faced this new issue.

**Comment by RaulPPelaez** [comment]
@rusty1s I think this should be closed.
